guide/src/project_layout.md

```diff
@@ -144,3 +144,17 @@ my-project
 └── src
     └── lib.rs
 ```
+
+## Data
+
+You can add wheel data by creating a `<module_name>.data` folder or setting its location as `data` in pyproject.toml under `[tool.maturin]` or in Cargo.toml under `[project.metadata.maturin]`.
```

Comment:
> You can add wheel data by creating a `<module_name>.data` folder

Does this mean that creating such folder will suddenly make a pure Rust project becoming a mixed Rust/Python project since it may create a `<python_module>` folder when `module_name == python_module`?

Reply:
mixed/pure and data are orthogonal, they are both options in `ProjectLayout`: https://github.com/PyO3/maturin/blob/5eb0fcb3655206a81ac6eb3a7c71a9450da23223/src/build_context.rs#L56-L67

By default, having a `<module_name>` dir will activate mixed mode, while having a `<module_name>.data` dir will activate "data mode" (btw i'm open to another default directory name, it's just what wheels are using internally and it seemed nice because it's not something people could have by accident like a pure `data` directory)

Reply:
The problem is that mixed mode also needs a `__init__.py`, so adding a `<module_name>.data` dir to a pure Rust mode will break the build if user forgot to add a `__init__.py`.

https://github.com/PyO3/maturin/pull/906/files#diff-87a24e61912b4f23e30d6031f5adc136adfa6008772880ae53a355cedfaee466R116-R118

Reply:
maybe i'm missing something, but i thought that's exactly the case i covered with the `with-data` test crate:

```
├── Cargo.lock
├── Cargo.toml
├── check_installed
│   └── check_installed.py
├── pyproject.toml
├── src
│   └── lib.rs
└── with_data.data
    ├── data
    │   └── data_subdir
    │       └── hello.txt
    └── headers
        └── empty.h
```

There you have no `__init__.py`, but a `with_data.data` and it works as pure-rust package

Reply:
Oh, great!